### PR TITLE
Prevent crash when selecting papenmeier braille display driver, remove brxcom wrapper

### PR DESCRIPTION
**Pull request by Stefan-Kliesch-FHP**
### Link to issue number:
None.

### Summary of the issue:
BrxCom wrapper is not compatible with Python 3 anymore and causes NVDA to crash.
Furthermore BrxCom is deprecated and is no longer supported.

### Description of how this pull request fixes the issue:
BrxCom support via wrapper dll removed from braille display driver. 

### Testing strategy:
Modified braille display driver tested via Developer Scratchpad.

### Known issues with pull request:
None.

### Change log entries:
Fixed a bug where selecting the Papenmeier Braille Display Driver caused NVDA to crash when BrxCom is installed.

- [X] Pull Request description:
  - description is up to date
  - change log entries
- [X] Testing:
  - Manual testing
- [X] API is compatible with existing add-ons.
- [X] Documentation:
  - User Documentation
- [X] UX of all users considered:
  - Braille

### CODE REVIEW

**Comment by Stefan-Kliesch-FHP** [comment]
This is my first pull request.
Please provide some help if something is not as it should be or as it be expected.

**Comment by lukaszgo1** [comment]
Thanks @Stefan-Kliesch-FHP That is very helpful to know. Could you please update the description of this PR to mention that BrxCom is deprecated and therefore support for it is removed? Also please remove mentions of BrxCom from user_docs\enuserGuide.t2t

**Comment by leonardder** [comment]
@Stefan-Kliesch-FHP Great to have you on board here! I'm glad to review the driver later this week!

**Comment by leonardder** [comment]
@Stefan-Kliesch-FHP Could you also elaborate on the papenmeier_serial driver? Do you think it is still used, or is it obsolete?

**Comment by Stefan-Kliesch-FHP** [comment]
@leonardder I think only a very few user will use it with the old braille dispays. I never tested it via serial port. 
Looking forward for your testing. 

I tested the braille driver with a BRAILLEX EL40c, BRAILLEX Trio, BRAILLEX Live+.
But only with USB connection. I will perform a bluetooth test later this week.

I removed the BrxCom from the english and german userGuide.t2t.

File: user_docs/de/userGuide.t2t
Comment (seanbudd) [inline user_docs/de/userGuide.t2t:2765]: Could you remove the changes from this file?
The German translation will need to be fixed by our translators via the translation system: https://github.com/nvaccess/nvda/wiki/TranslatingUserGuide

File: source/brailleDisplayDrivers/papenmeier.py
Comment (seanbudd) [inline source/brailleDisplayDrivers/papenmeier.py:142]: Can we fix the missing space and wording here?

```suggestion
		"""Try to connect to usb device, this is triggered when bluetooth
```

**Review by seanbudd — COMMENTED** [review]
Thanks for this PR @Stefan-Kliesch-FHP.
I've suggested 2 minor changes, I think this should be good to merge after these are fixed up, but I'll wait for @leonardder to do the review he wants to do.

File: user_docs/de/userGuide.t2t
Comment (Stefan-Kliesch-FHP) [reply user_docs/de/userGuide.t2t]: should be restored!

File: source/brailleDisplayDrivers/papenmeier.py
Comment (Stefan-Kliesch-FHP) [reply source/brailleDisplayDrivers/papenmeier.py]: I committed the suggesttion. Or didn't I?
Why is this marked as "outdated"?

File: user_docs/de/userGuide.t2t
Comment (lukaszgo1) [inline user_docs/de/userGuide.t2t:3275]: Please reintroduce this line - there should be no changes to the German documentation in this PR as it would introduce conflicts for translators.

File: user_docs/en/userGuide.t2t
Comment (lukaszgo1) [inline user_docs/en/userGuide.t2t:2863]: Just to make sure - since you've removed this line are you saying that braille input is supported with other devices as well?

Comment (Stefan-Kliesch-FHP) [reply user_docs/en/userGuide.t2t:2863]: BrxCom is a kind of connection chain. BrxCom connects to the braille display and the screenreader connects to BrxCom.
All keyboard inputs are handled by BrxCom and not passed through to the screenreader. NVDA is able to handle the qwerty and the braille input itself. Other screenreaders also implemented this function, so BrxCom is not needed anymore.

File: user_docs/de/userGuide.t2t
Comment (lukaszgo1) [reply user_docs/de/userGuide.t2t:3275]: I'm afraid it isn't. This should be just empty line not hyphen. Essentially when looking at the list of files modified by this PR German user guide should not be mentioned at all.

Comment (Stefan-Kliesch-FHP) [reply user_docs/de/userGuide.t2t]: So I restored it from nvaccess:master and it's gone from the "files changed" list!

**Comment by seanbudd** [comment]
How did the bluetooth tests go? This looks ready to merge otherwise.

**Comment by Stefan-Kliesch-FHP** [comment]
Sorry, been very busy last week. I will test it, today or tomorrow.

**Comment by Stefan-Kliesch-FHP** [comment]
So I tested it immediately, but I had a problem with the "Build (for testing PR)" version. The settings dialog wasn't available.
So I tested it with the 2021.3.1 and added the papenmeier.py to the developer scratchpad. Which worked fine with a bluetooth connection to the BRAILLEX Live+. Keyboard inputs work also fine. 
Any ideas why I couldn't use the the testbuild?

**Comment by Stefan-Kliesch-FHP** [comment]
I encoutern an other strange behavior. If there was a working bluetooth connection via NVDA and the BRAILLEX Live+ and I switch off the braille display, NVDA starts to act strange. After a few seconds it lags with keyboard inputs and GUI updates. And it's not recovering. I have to switch on the braille display again or end the NVDA task in the Task-Manager. I think it's due to the lost connection.

**Comment by Stefan-Kliesch-FHP** [comment]
Hello everyone,
I was able to test it with the PR Build. 
I have to press Ctrl + NVDA + A to selecct the braille display.
And it works so far, but the bluetooth disconnect problem remains. 
It seems that either the HWPortUtils nor the Serial-object throw an exception when the bluetooth device is disconnected. 
Any ideas about that?

When I try to select the braille display via the main settings dialog I get thje following traceback:
-Base configuration saved
-ERROR - unhandled exception (12:21:42.928) - MainThread (21620):
- Traceback (most recent call last):
-  File "gui\__init__.pyc", line 224, in onNVDASettingsCommand
-  File "gui\__init__.pyc", line 166, in _popupSettingsDialog
-  File "gui\settingsDialogs.pyc", line 463, in __init__
-  File "gui\settingsDialogs.pyc", line 193, in __init__
-  File "gui\settingsDialogs.pyc", line 4000, in makeSettings
-  File "gui\settingsDialogs.pyc", line 532, in makeSettings
-  File "gui\settingsDialogs.pyc", line 641, in _doCategoryChange
-  File "gui\settingsDialogs.pyc", line 569, in _getCategoryPanel
-  File "gui\settingsDialogs.pyc", line 327, in __init__
-  File "gui\settingsDialogs.pyc", line 337, in _buildGui
-  File "gui\settingsDialogs.pyc", line 805, in makeSettings
-  File "config\__init__.pyc", line 391, in getStartOnLogonScreen
-UnboundLocalError: local variable 'k' referenced before assignment
-INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (12:21:44.871) - MainThread (21620):

This can't be caused by the papenmeier driver, so any ideas?

**Comment by leonardder** [comment]
Hey @Stefan-Kliesch-FHP,

To get rid of the issues with bluetooth, I'd suggest a more thorough rewrite of the driver that is more like the other native drivers (e.g. Handy Tech, Hims, Baum). In short, make use of the hwIo module to ensure asynchronous communication with the display. Unfortunately, the FTDI2 module is a bit of a show stopper here, since it doesn't support overlapped IO I believe.
Now I wonder, is it really necessary to communicate with the device with this FTDI2 module or is it also possible to use generic Win32 serial communication somehow? I once tried to get the Win32 handle from the FTDI2 handle in order to write to it with the Win32 Serial APIs, but that was slightly hacky. How do you communicate with these display with the other screen readers?

**Comment by leonardder** [comment]
I noticed that the FTDI driver should support both D2XX and virtual COM port access. So if we can switch to the virtual COM port model somehow, that would be greatly preferred and bring the following benefits. Note that these things shouldn't be impossible with D2XX, it will just take much more research and effort to get it working:

1. Implementing braille display auto detection.
2. Unifying the bluetooth and usb code as both use virtual COM ports
3. Probably better performance because we can easily switch to async I/O.

**Comment by Stefan-Kliesch-FHP** [comment]
@leonardder 
This would mostly effect the USB connections, which works fine so far.
I would suggest that we just merge this pull request. 
And if I find some time, I will consider to use HWio for the Bluetooth communication in the future. Including a new PR.

**Review by seanbudd — APPROVED** [review]
Thanks @Stefan-Kliesch-FHP

**Review by Qchristensen — APPROVED** [review]
Great work, thank you!

**Pull request merged**
